editor/gui/editor_bottom_panel.cpp

```diff
@@ -59,6 +76,34 @@ void EditorBottomPanel::_switch_by_control(bool p_visible, Control *p_control) {
 	}
 }
 
+void EditorBottomPanel::_scroll(bool p_right) {
+	HScrollBar *h_scroll = button_scroll->get_h_scroll_bar();
+	if (Input::get_singleton()->is_key_pressed(Key::CTRL)) {
+		h_scroll->set_value(p_right ? h_scroll->get_max() : 0);
+	} else if (Input::get_singleton()->is_key_pressed(Key::SHIFT)) {
+		h_scroll->set_value(h_scroll->get_value() + (h_scroll->get_page() * (p_right ? 1 : -1)));
```

Comment:
```suggestion
		h_scroll->set_value(h_scroll->get_value() + h_scroll->get_page() * (p_right ? 1 : -1));
```
Not needed due to operator precedence.
Same below.